### PR TITLE
Using `PolicyTargetReference` instead of `Selector`

### PR DESCRIPTION
**Pull request by didierofrivia**
In order to **not** rely on Gateway a̶n̶n̶o̶t̶a̶t̶i̶o̶n̶s̶ labels/selectors, we are now using [PolicyTargetReference](https://istio.io/latest/docs/reference/config/type/workload-selector/#PolicyTargetReference). This fixes the issue when no annotation has been added to the managed gateway (or removed/updated).

**Verification steps**

1. Set up the cluster
```sh
make local-setup
```

2. Remove the deployed gw label
```sh
kubectl -n istio-system label gateway istio-ingressgateway istio-
```

3. Apply Kuadrant CR
```sh
kubectl -n kuadrant-system apply -f - <<EOF
apiVersion: kuadrant.io/v1beta1
kind: Kuadrant
metadata:
  name: kuadrant
spec: {}
EOF
```

4. Deploy ToyStore API and route the traffic
```sh
kubectl apply -f examples/toystore/toystore.yaml

kubectl apply -f - <<EOF
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: toystore
spec:
  parentRefs:
  - name: istio-ingressgateway
    namespace: istio-system
  hostnames:
  - api.toystore.com
  rules:
  - matches:
    - path:
        type: Exact
        value: "/toy"
      method: GET
    backendRefs:
    - name: toystore
      port: 80
EOF
```

5. Apply a RLP
```sh
kubectl apply -f - <<EOF
apiVersion: kuadrant.io/v1beta2
kind: RateLimitPolicy
metadata:
  name: toystore
spec:
  targetRef:
    group: gateway.networking.k8s.io
    kind: HTTPRoute
    name: toystore
  limits:
    "default":
      rates:
      - limit: 5
        duration: 10
        unit: second
EOF
```

6. Check the envoy config for the wasm shim configuration for the RLP

```sh
kubectl port-forward --namespace istio-system deployment/istio-ingressgateway-istio 15000:15000 &

curl -v "<http://127.0.0.1:15000/config_dump?format=json>" | yq e -P -
```

It should contain
```sh
...
 config:
              name: istio-system.kuadrant-istio-ingressgateway
              vm_config:
                runtime: envoy.wasm.runtime.v8
                code:
                  local:
                    filename: /var/lib/istio/data/81221938ebcbc4550eb35f72aac65d0939d89335832b5a022226fb2526806e9e/9b28356942b98c2a73dd1428403ec45836f956114208dfceb88f8f957e94e45d.wasm
              configuration:
                '@type': type.googleapis.com/google.protobuf.StringValue
                value: '{"failureMode":"deny","rateLimitPolicies":[{"domain":"default/toystore","hostnames":["api.toystore.com"],"name":"default/toystore","rules":[{"conditions":[{"allOf":[{"operator":"eq","selector":"request.url_path","value":"/toy"},{"operator":"eq","selector":"request.method","value":"GET"}]}],"data":[{"static":{"key":"limit.default__37a8eec1","value":"1"}}]}],"service":"kuadrant-rate-limiting-service"}]}'
...
```

### CODE REVIEW

**Comment by didierofrivia** [comment]
I haven't removed `WorkloadSelectorFromGateway` since it's still used by https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/authpolicy_istio_authorizationpolicy.go#L98 and https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/limitador_cluster_envoyfilter_controller.go#L120

**Review by ehearneRedHat — APPROVED** [review]
Looks good to merge - request did contain

```json
config:
              name: istio-system.kuadrant-istio-ingressgateway
              vm_config:
                runtime: envoy.wasm.runtime.v8
                code:
                  local:
                    filename: /var/lib/istio/data/81221938ebcbc4550eb35f72aac65d0939d89335832b5a022226fb2526806e9e/9b28356942b98c2a73dd1428403ec45836f956114208dfceb88f8f957e94e45d.wasm
              configuration:
                '@type': type.googleapis.com/google.protobuf.StringValue
                value: '{"failureMode":"deny","rateLimitPolicies":[{"domain":"default/toystore","hostnames":["api.toystore.com"],"name":"default/toystore","rules":[{"conditions":[{"allOf":[{"operator":"eq","selector":"request.url_path","value":"/toy"},{"operator":"eq","selector":"request.method","value":"GET"}]}],"data":[{"static":{"key":"limit.default__37a8eec1","value":"1"}}]}],"service":"kuadrant-rate-limiting-service"}]}'
```

File: pkg/istio/utils.go [hunk 1]
```diff
@@ -22,3 +22,11 @@ func WorkloadSelectorFromGateway(ctx context.Context, k8sClient client.Client, g
 		MatchLabels: gatewayWorkloadSelector,
 	}
 }
+
+func PolicyTargetRefFromGateway(gateway *gatewayapiv1.Gateway) *istiocommon.PolicyTargetReference {
+	return &istiocommon.PolicyTargetReference{
+		Group: "gateway.networking.k8s.io",
```
Comment (eguzki) [inline pkg/istio/utils.go:28]:
```suggestion
		Group: gatewayapiv1.GroupName,
```

**Review by eguzki — APPROVED** [review]

**Comment by guicassolato** [comment]
What does this have to do with gateway annotations?

**Comment by eguzki** [comment]
> What does this have to do with gateway annotations?

When the gateway does not have labels (nothing to do with annotations), the label selector implemented by kuadrant is failing as it is empty. Empty label selector will not match any gateway, thus rate limiting not enforced.

**Comment by guicassolato** [comment]
> > What does this have to do with gateway annotations?
> 
> When the gateway does not have labels (nothing to do with annotations), the label selector implemented by kuadrant is failing as it is empty. Empty label selector will not match any gateway, thus rate limiting not enforced.

Thanks @eguzki. That's what I figured. Maybe fix in the description of the PR, @didierofrivia?

**Pull request merged**